### PR TITLE
[4657] Reactify Global App (Log Console)

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/4657

### CODE REVIEW

File: ui/app/src/components/ClustersManagement/ClustersManagement.tsx [hunk 1]
```diff
@@ -96,7 +96,7 @@ export default function ClustersManagement() {
   return (
     <section>
       <GlobalAppToolbar
-        title={<FormattedMessage id="GlobalMenu.Cluster" defaultMessage="Cluster" />}
+        title={<FormattedMessage id="globalMenu.cluster" defaultMessage="Cluster" />}
```
Comment (rart) [inline ui/app/src/components/ClustersManagement/ClustersManagement.tsx]: Didn't we have all of these with capitalization on the `G`? Why did you change it? Also why you changing clusters at all?

File: ui/app/src/components/GroupsManagement/GroupsManagement.tsx [hunk 2]
```diff
@@ -103,7 +103,7 @@ export default function GroupsManagement() {
   return (
     <section>
       <GlobalAppToolbar
-        title={<FormattedMessage id="GlobalMenu.Groups" defaultMessage="Groups" />}
+        title={<FormattedMessage id="globalMenu.groups" defaultMessage="Groups" />}
```
Comment (rart) [inline ui/app/src/components/GroupsManagement/GroupsManagement.tsx]: Same as comment as above

File: ui/app/src/components/UsersManagement/UsersManagement.tsx [hunk 3]
```diff
@@ -106,7 +106,7 @@ export default function UsersManagement(props: UsersManagementProps) {
   return (
     <section>
       <GlobalAppToolbar
-        title={<FormattedMessage id="GlobalMenu.Users" defaultMessage="Users" />}
+        title={<FormattedMessage id="globalMenu.users" defaultMessage="Users" />}
```
Comment (rart) [inline ui/app/src/components/UsersManagement/UsersManagement.tsx]: same

File: ui/app/src/components/TokenManagement/TokenManagement.tsx [hunk 4]
```diff
@@ -315,7 +315,7 @@ export default function TokenManagement() {
   return (
     <section>
       <GlobalAppToolbar
-        title={<FormattedMessage id="GlobalMenu.TokenManagement" defaultMessage="Token Management" />}
+        title={<FormattedMessage id="globalMenu.tokenManagement" defaultMessage="Token Management" />}
```
Comment (rart) [inline ui/app/src/components/TokenManagement/TokenManagement.tsx:318]: same

File: ui/app/src/components/PluginManagement/PluginManagement.tsx [hunk 5]
```diff
@@ -268,7 +268,7 @@ export const PluginManagement = (props: PluginManagementProps) => {
           {plugins?.length === 0 && (
             <EmptyState
               title={
-                <FormattedMessage id="PluginManagement.emptyList" defaultMessage="There are not plugins installed" />
+                <FormattedMessage id="pluginManagement.emptyList" defaultMessage="There are not plugins installed" />
```
Comment (rart) [inline ui/app/src/components/PluginManagement/PluginManagement.tsx]: same

File: ui/app/src/components/PluginManagement/PluginManagement.tsx [hunk 6]
```diff
@@ -234,7 +234,7 @@ export const PluginManagement = (props: PluginManagementProps) => {
                   </StyledTableCell>
                   <StyledTableCell align="left">
                     <Typography variant="subtitle2">
-                      <FormattedMessage id="PluginManagement.installationDate" defaultMessage="Installation Date" />
+                      <FormattedMessage id="pluginManagement.installationDate" defaultMessage="Installation Date" />
```
Comment (rart) [inline ui/app/src/components/PluginManagement/PluginManagement.tsx:237]: same

File: ui/app/src/components/PluginManagement/PluginManagement.tsx [hunk 7]
```diff
@@ -201,7 +201,7 @@ export const PluginManagement = (props: PluginManagementProps) => {
         <EmptyState
           title={
             <FormattedMessage
-              id="PluginManagement.listPluginPermission"
+              id="pluginManagement.listPluginPermission"
```
Comment (rart) [inline ui/app/src/components/PluginManagement/PluginManagement.tsx:204]: same

File: ui/app/src/components/PluginManagement/PluginManagement.tsx [hunk 8]
```diff
@@ -191,7 +191,7 @@ export const PluginManagement = (props: PluginManagementProps) => {
         disabled={permissions === null || listPluginsPermission === false}
       >
         {installPluginsPermission ? (
-          <FormattedMessage id="PluginManagement.searchPlugin" defaultMessage="Search & install" />
+          <FormattedMessage id="pluginManagement.searchPlugin" defaultMessage="Search & install" />
```
Comment (rart) [inline ui/app/src/components/PluginManagement/PluginManagement.tsx:194]: same

File: ui/app/src/components/PluginManagement/PluginManagement.tsx [hunk 9]
```diff
@@ -179,7 +179,7 @@ export const PluginManagement = (props: PluginManagementProps) => {
       {!embedded && (
         <>
           <Typography variant="h4" component="h1" className={classes.title}>
-            <FormattedMessage id="PluginManagement.title" defaultMessage="Plugin Management" />
+            <FormattedMessage id="pluginManagement.title" defaultMessage="Plugin Management" />
```
Comment (rart) [inline ui/app/src/components/PluginManagement/PluginManagement.tsx]: same

File: ui/app/src/components/LoggingLevelsManagement/LoggingLevelsManagement.tsx [hunk 10]
```diff
@@ -91,7 +91,7 @@ export default function LoggingLevelsManagement() {
   return (
     <section>
       <GlobalAppToolbar
-        title={<FormattedMessage id="GlobalMenu.LoggingLevelsEntryLabel" defaultMessage="Logging Levels" />}
+        title={<FormattedMessage id="globalMenu.loggingLevelsEntryLabel" defaultMessage="Logging Levels" />}
```
Comment (rart) [inline ui/app/src/components/LoggingLevelsManagement/LoggingLevelsManagement.tsx:94]: same

File: ui/app/src/components/LogConsoleDetailsDialog/LogConsoleDetailsDialog.tsx [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+export default function LogConsoleDetailsDialog(props: LogConsoleDetailsDialogProps) {
+  const { open, onClose } = props;
+  return (
+    <Dialog open={open} onClose={onClose} fullWidth maxWidth="md">
+      <LogConsoleDetailsDialogUI {...props} />
+    </Dialog>
+  );
+}
+
+function LogConsoleDetailsDialogUI(props: LogConsoleDetailsDialogProps) {
```
Comment (rart) [inline ui/app/src/components/LogConsoleDetailsDialog/LogConsoleDetailsDialog.tsx]: Move to own file

File: ui/app/src/components/LogConsoleDetailsDialog/LogConsoleDetailsDialog.tsx [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+  const { logEvent, onClose, onClosed } = props;
+  const localeBranch = useSelection((state) => state.uiConfig.locale);
+  const classes = useStyles();
+
+  useMount(() => {
+    return () => {
+      onClosed?.();
+    };
+  });
```
Comment (rart) [inline ui/app/src/components/LogConsoleDetailsDialog/LogConsoleDetailsDialog.tsx]: useUnmount instead

File: ui/app/src/components/LogConsoleDetailsDialog/LogConsoleDetailsDialog.tsx [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+          </Box>
+        </Box>
+      </DialogBody>
+    </>
+  );
+}
+//
+// width: '100%',
+//   display: 'flex',
+//   alignItems: 'center'
```
Comment (rart) [inline ui/app/src/components/LogConsoleDetailsDialog/LogConsoleDetailsDialog.tsx]: x

File: ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+              </Typography>
+            </GlobalAppGridCell>
+            <GlobalAppGridCell align="left" className="bordered padded10 width25">
+              <Typography variant="subtitle2">
+                <FormattedMessage id="words.state" defaultMessage="Timestamp" />
+              </Typography>
+            </GlobalAppGridCell>
+            <GlobalAppGridCell align="left" className="bordered padded10 width25">
+              <Typography variant="subtitle2">
+                <FormattedMessage id="words.url" defaultMessage="Thread" />
```
Comment (rart) [inline ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx]: Check id

File: ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+  const localeBranch = useSelection((state) => state.uiConfig.locale);
+
+  return (
+    <TableContainer>
+      <Table>
+        <TableHead>
+          <GlobalAppGridRow className="hoverDisabled">
+            <GlobalAppGridCell align="left" className="bordered padded10 width10">
+              <Typography variant="subtitle2">
+                <FormattedMessage id="clusterGrid.localAddress" defaultMessage="Level" />
```
Comment (rart) [inline ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx]: Check id

File: ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+              </Typography>
+            </GlobalAppGridCell>
+            <GlobalAppGridCell align="left" className="bordered padded10 width25">
+              <Typography variant="subtitle2">
+                <FormattedMessage id="words.url" defaultMessage="Thread" />
+              </Typography>
+            </GlobalAppGridCell>
+            <GlobalAppGridCell align="left" className="bordered padded10 width20">
+              <Typography variant="subtitle2">
+                <FormattedMessage id="clusterGrid.remoteName" defaultMessage="Site" />
```
Comment (rart) [inline ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx]: Check id

File: ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+              </Typography>
+            </GlobalAppGridCell>
+            <GlobalAppGridCell align="left" className="bordered width60 padded10">
+              <Typography variant="subtitle2">
+                <FormattedMessage id="clusterGrid.authType" defaultMessage="Message" />
```
Comment (rart) [inline ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx]: Check id

File: ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+              </Typography>
+            </GlobalAppGridCell>
+            <GlobalAppGridCell align="left" className="bordered padded10 width10">
+              <FormattedMessage id="clusterGrid.authType" defaultMessage="Details" />
```
Comment (rart) [inline ui/app/src/components/LogConsoleGrid/LogConsoleGridUI.tsx]: Check id

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/GroupsManagement/GroupsManagement.tsx (hunk 2 above)
Comment (aachavarria) [reply ui/app/src/components/GroupsManagement/GroupsManagement.tsx]: in i18n-legacy.ts we have `home.globalMenu.groups`, I thought we were using GlobalMenu.Groups. That is why I changed all of them, we can use the same as i18n-legacy.ts or keep it using globalMenu.*

File: ui/app/src/components/PluginManagement/PluginManagement.tsx (hunk 5 above)
Comment (aachavarria) [reply ui/app/src/components/PluginManagement/PluginManagement.tsx]: the syntax we are using of all  FormattedMessage is with the first letter in lowercase

**Review by rart — APPROVED** [review]

**Comment by sumerjabri** [comment]
@aachavarria please fix conflicts

**Review by rart — CHANGES_REQUESTED** [review]
Aaron, as discussed, Log Console should just be called LogConsole, not LogConsoleManagement.

**Review by rart — APPROVED** [review]
